src/test/compile-fail/issue-15524.rs

```diff
@@ -12,13 +12,16 @@ const N: isize = 1;
 
 enum Foo {
     A = 1,
-    B = 1, //~ ERROR discriminant value `1isize` already exists
-    //~^^ NOTE conflicting
+    B = 1, //~ ERROR discriminant value
+    //~^ NOTE enum already
+    //~^^^ NOTE first use
```

Comment:
Assuming I'm reading this right, the `NOTE first use` for these errors should be on a label on line 14 `A = 1`.

Are the spans you're getting showing up on 15, 18, and 20 instead?